content/blog/2024/04/sourceless-ck-springboot/index.md

```diff
@@ -0,0 +1,135 @@
+
+> The entire building and publishing process can be substituted by a more formal CICD process. For simplicity reason we've preferred the local prototyping approach.
+
+### Create the application
+
+Let's create a Java application with a single endpoint:
+
+```java
+import org.apache.camel.builder.RouteBuilder;
+
+public class PlatformHttpServer extends RouteBuilder {
+  @Override
+  public void configure() throws Exception {
+    from("platform-http:/hello?httpMethodRestrict=GET").setBody(simple("Hello ${header.name}"));
+  }
+}
+```
+
+Thanks to Camel JBang, we can quickly run it locally and validate the prototype:
+
+```
+camel run PlatformHttpServer.java
+```
+
+In a separate shell:
+
+```
+$ curl localhost:8080/hello --header 'name: world'
+Hello world
+```
+
+It works good enough to proceed to next step.
+
+### Publish the application
+
+Let's leverage the Camel JBang capabilities now to export the application in our runtime of choice, say, Camel Spring Boot:
+
+```
+$ camel export PlatformHttpServer.java --runtime spring-boot --gav org.acme:my-csb:1.0.0 --dir csb
+```
+
+In the `csb` directory, we'll have the exported Maven project. Let's build and run this as well. We should expect the same result, but, this time, with a Spring Boot application.
+
+```
+$ mvn clean package
+...
+$ java -jar target/my-csb-1.0.0.jar
+...
+2024-04-02T12:49:38.492+02:00  INFO 547806 --- [           main] org.acme.mycsb.CamelApplication          : Started CamelApplication in 3.684 seconds (process running for 4.121)
+...
+$ curl localhost:8080/hello --header 'name: world'
+Hello world
+```
+
+As we have no surprises, let's put this into a container and publish the the same registry used by Camel K on Kubernetes. We need to create a Dockerfile:
+
+```
+FROM eclipse-temurin:17
+COPY --chown=185 target/my-csb-1.0.0.jar /deployments/my-camel-app.jar
+CMD java -jar /deployments/my-camel-app.jar
+```
+
+We can now use this manifest to build and push the container to our registry (in my case a local registry hosted on 10.103.186.147):
+
+```
+docker build -t 10.103.186.147/default/csb .
+docker push 10.103.186.147/default/csb
+```
+
+### Operate the application
+
+We are now in the position to operate the application with Camel K. The goal is to take care of the deployment. As this is a Knative, it will run the related traits required to run a `KnativeService` instead of a regular `Deployment` and scale to 0 when there is no traffic:
+
+```
+$ kamel run --image 10.103.186.147/default/csb
+...
+[1] 2024-04-02T11:00:14.950Z  INFO 7 --- [           main] org.acme.mycsb.CamelApplication          : Started CamelApplication in 3.239 seconds (process running for 3.7)
+```
+
+We can now start **monitoring the applications** and see if it really scale to 0 after some time of inactivity:
+
+```
+$ kubectl get it -w
+NAME   PHASE     RUNTIME PROVIDER   RUNTIME VERSION   KIT       REPLICAS
+csb    Running   quarkus            3.2.3             kit-csb   1
```

Comment:
Here is says quarkus - I guess this can be confusing when you say that its a spring boot app.

Reply:
Yeah, I noticed that too and I created an issue to fix it soon: https://github.com/apache/camel-k/issues/5309 - do you think is it fine if we just remove from the post?